src/mol-theme/substance.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2021 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Alexander Rose <alexander.rose@weirdbyte.de>
+ */
+
+import { Loci } from '../mol-model/loci';
+import { Structure, StructureElement } from '../mol-model/structure';
+import { Script } from '../mol-script/script';
+import { Material } from '../mol-util/material';
+
+export { Substance };
+
+type Substance = { readonly layers: ReadonlyArray<Substance.Layer> }
+
+function Substance(layers: ReadonlyArray<Substance.Layer>): Substance {
+    return { layers };
+}
+
+namespace Substance {
+    export type Layer = { readonly loci: StructureElement.Loci, readonly material: Material, readonly clear: boolean }
+    export const Empty: Substance = { layers: [] };
+
+    export function areEqual(sA: Substance, sB: Substance) {
+        if (sA.layers.length === 0 && sB.layers.length === 0) return true;
+        if (sA.layers.length !== sB.layers.length) return false;
```

Comment:
All the functions here are almost identical to Overpaint and Transparency as far as the loci interactions go.

How about adding a ``LociLayerHelper`` which handles the loci interactions?

E.g. 

```ts
type LociLayer = { readonly loci: StructureElement.Loci }
function remapLociLayers<T extends LociLayer>(layers: ReadonlyArray<T>, structure: Structure): ReadonlyArray<T> {
  // ...
      layers.push({ ...layer, loci: StructureElement.Loci.remap(loci, structure) });
  // ...
}
```

Reply:
Yes, made #293 so we don't forget